sycl/include/sycl/sycl_span.hpp

```diff
@@ -217,9 +285,8 @@ template <typename _Tp, size_t _Extent> class _SYCL_SPAN_TEMPLATE_VIS span {
   constexpr span(const span &) noexcept = default;
   constexpr span &operator=(const span &) noexcept = default;
 
-  template <size_t _Sz = _Extent>
-  _SYCL_SPAN_INLINE_VISIBILITY constexpr explicit span(
-      element_type (&__arr)[_Sz])
+  template <size_t _Sz>
+  _SYCL_SPAN_INLINE_VISIBILITY constexpr span(element_type (&__arr)[_Sz])
       : __data{__arr} {
     (void)_Sz;
     _SYCL_SPAN_ASSERT(_Extent == _Sz,
```

Comment:
1) This should be a compile-time check (or maybe even SFINAE, depending on the spec wording).
2) Dynamic extent on the span should be allowed if I'm reading https://en.cppreference.com/w/cpp/container/span/span.html correctly.

I'd guess same for other size checks below.